src/core/lib/security/authorization/authorization_engine.cc

```diff
@@ -74,4 +74,77 @@ AuthorizationEngine::AuthorizationEngine(
+          "url_path",
+          google::api::expr::runtime::CelValue::CreateStringView(args.path()));
+    } else if (elem == "host") {
+      activation->InsertValue(
+          "host",
+          google::api::expr::runtime::CelValue::CreateStringView(args.host()));
+    } else if (elem == "method") {
+      activation->InsertValue(
+          "method", google::api::expr::runtime::CelValue::CreateStringView(
+                        args.method()));
+    } else if (elem == "headers") {
+      std::multimap<absl::string_view, absl::string_view> headers =
+          args.headers();
+      std::vector<std::pair<google::api::expr::runtime::CelValue,
+                            google::api::expr::runtime::CelValue>>
+          header_items;
+      for (const auto& header_key : header_keys_) {
+        auto header_item = headers.find(header_key);
+        if (header_item != headers.end()) {
+          header_items.push_back(
+              std::pair<google::api::expr::runtime::CelValue,
+                        google::api::expr::runtime::CelValue>(
+                  google::api::expr::runtime::CelValue::CreateStringView(
+                      header_key),
+                  google::api::expr::runtime::CelValue::CreateStringView(
+                      header_item->second)));
+        }
+      }
+      headers_ = google::api::expr::runtime::ContainerBackedMapImpl::Create(
+          absl::Span<std::pair<google::api::expr::runtime::CelValue,
+                               google::api::expr::runtime::CelValue>>(
+              header_items));
+      activation->InsertValue(
+          "headers",
+          google::api::expr::runtime::CelValue::CreateMap(headers_.get()));
+    } else if (elem == "source_address") {
+      activation->InsertValue(
+          "source_address",
+          google::api::expr::runtime::CelValue::CreateStringView(
+              args.sourceAddress()));
+    } else if (elem == "source_port") {
+      activation->InsertValue(
+          "source_port",
+          google::api::expr::runtime::CelValue::CreateInt64(args.sourcePort()));
+    } else if (elem == "destination_address") {
+      activation->InsertValue(
+          "destination_address",
+          google::api::expr::runtime::CelValue::CreateStringView(
+              args.destinationAddress()));
+    } else if (elem == "destination_port") {
+      activation->InsertValue("destination_port",
+                              google::api::expr::runtime::CelValue::CreateInt64(
+                                  args.destinationPort()));
+    } else if (elem == "spiffe_id") {
+      activation->InsertValue(
+          "spiffe_id", google::api::expr::runtime::CelValue::CreateStringView(
+                           args.spiffeID()));
+    } else if (elem == "cert_server_name") {
+      activation->InsertValue(
+          "cert_server_name",
+          google::api::expr::runtime::CelValue::CreateStringView(
+              args.certServerName()));
+    }
```

Comment:
Add an else too, and log error
It would mean the policy has an attribute that is not supported by us.